### ISSUE TITLE
Access to UserContext inside handler

### ISSUE THREAD

**Issue by hfroliveira** [graphql-dotnet/graphql-dotnet#455]
Hello,

I want to congratulate joemcbride and everyone contributing for this great project. 

I would like to know if there is a way to access UserContext from inside resolver methods while using Handler/Endpoints
approach.

For instance, using the StarWars sample, I would like to access UserContext (passed in the ExecuteAsync call) inside GetHero method.

```csharp
public class Query
{
  [GraphQLMetadata("hero")]
  public Droid GetHero()
  {
    return new Droid { Id = "123", Name = "R2-D2" };
  }
}
```

Thanks.

**Comment by joemcbride** [comment]
It doesn't support UserContext directly at present, but you can add `ResolveFieldContext` as a parameter.

```csharp
public class Query
{
  [GraphQLMetadata("hero")]
  public Droid GetHero(ResolveFieldContext context)
  {
    var userContext = context.UserContext;
    return new Droid { Id = "123", Name = "R2-D2" };
  }
}
```

**Comment by hfroliveira** [comment]
Hello,

I tried the solution but I got the following exception:  Index was outside the bounds of the array.

Inspecting the latest code it looks a problem at Resolve method (MethodModelBinderResolver.cs, line:38)
while checking if the type of the argument matches the type of the data source type.

I was able to proceed with the following fix:

`if ((_parameters.Length != index) && (context.Source?.GetType() == _parameters[index].ParameterType))`

I've attached the code of my sample program.

Thanks.
[Program.zip](https://github.com/graphql-dotnet/graphql-dotnet/files/1370006/Program.zip)
